src/librustc_trans/back/msvc/mod.rs

```diff
@@ -191,7 +214,7 @@ mod platform {
                 .map(|dir| dir.path()).collect();
             dirs.sort();
             dirs.into_iter().rev().filter(|dir| {
-                dir.join("um").is_dir()
+                dir.join("um").join("x64").join("kernel32.lib").is_file()
```

Comment:
One person had a `um` directory which actually had libraries inside of it, but they weren't the libraries we needed. By checking for `kernel32.lib` specifically it would fix that situation by detecting it as invalid and falling back to another SDK (such as the Windows 8.1 SDK which that person did have). I made it check for `x64` because all the Windows SDKs we support have all the architectures anyway and I didn't feel the need to do it based on the architecture that was being requested.